### ISSUE TITLE
Problem with network scan

### ISSUE THREAD

**Issue by Zerbaib** [jokob-sk/NetAlertX#829]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues and I checked the docs https://github.com/jokob-sk/NetAlertX/tree/main/docs

### Current Behavior

I have some problem with network scanning:
- i have 
![image](https://github.com/user-attachments/assets/ca63f3f0-3741-4b16-90f2-2e0779b31cb5)
- but only the `192.168.0.0/24` was scanned
![image](https://github.com/user-attachments/assets/f1597e2c-0f68-40da-bb4f-3855c1beea42)
- the sys info for the thernet connection is
![image](https://github.com/user-attachments/assets/22377e08-dc50-477f-a89c-92caf057c8d6)

so the config is normaly good

### Expected Behavior

i search all my network device wen my router scan i can have
![image](https://github.com/user-attachments/assets/14b26f7b-2671-4caf-aaf3-4e75c23196c3)

but no on the netalertx

### docker-compose.yml

```python
docker run -d --rm --network=host \
  -v /root/netalert/config:/app/config \
  -v /root/netalert/db:/app/db \
  -e TZ=Europe/Berlin \
  -e PORT=20211 \
  jokobsk/netalertx:latest
```

### What branch are you running?

Production

### app.log

[app.log](https://github.com/user-attachments/files/17307586/app.log)

### Debug enabled

- [X] I have read and followed the steps in the wiki link above and provided the required debug logs and the log section covers the time when the issue occurs.

**Comment by jokob-sk** [comment]
Hi,

Can you confirm  you read this guide? https://github.com/jokob-sk/NetAlertX/blob/main/docs/SUBNETS.md \

Especially this section:

> ARPSCAN can scan multiple networks if the network allows it. To scan networks directly, the subnets must be accessible from the network where NetAlertX is running. This means NetAlertX needs to have access to the interface attached to that subnet. You can verify this by running the following command in the container:

> `sudo arp-scan --interface=eth0 192.168.1.0/24`

If so, can you please post sample output of the above call from your terminal using some of your subnets?

Could you please confirm that all subnets are on the same interface?

**Comment by MimaHack** [comment]
hello @jokob-sk  i'm facing the same probleme and when i run this sudo arp-scan --interface=eth0 192.168.1.0/24 it's work in the terminal but not on netalertx , what can be the solution for thsi ?

**Comment by jokob-sk** [comment]
@MimaHack Please open a separate issue and submit app logs showing your ARPCAN run so I don't confuse your setups. Please see this doc to know what data to submit: https://github.com/jokob-sk/NetAlertX/blob/main/docs/DEBUG_PLUGINS.md

**Comment by Zerbaib** [comment]
> Hi,
> 
> Can you confirm you read this guide? https://github.com/jokob-sk/NetAlertX/blob/main/docs/SUBNETS.md \
> 
> Especially this section:
> 
> > ARPSCAN can scan multiple networks if the network allows it. To scan networks directly, the subnets must be accessible from the network where NetAlertX is running. This means NetAlertX needs to have access to the interface attached to that subnet. You can verify this by running the following command in the container:
> 
> > `sudo arp-scan --interface=eth0 192.168.1.0/24`
> 
> If so, can you please post sample output of the above call from your terminal using some of your subnets?
> 
> Could you please confirm that all subnets are on the same interface?

yes i have read the guide
and the output is:
![image](https://github.com/user-attachments/assets/af4a8213-700b-4b98-9b73-c9777307283d)

**Comment by jokob-sk** [comment]
Thanks for that. The above output means the interface is incorrect or your host isn't able to access the given subnet. Simply since the above doesn't return results, NetAlertX can't return results either. That means the following from the above guide applies:

> If direct scans are not possible, you can use [supplementing plugins](https://github.com/jokob-sk/NetAlertX/blob/main/front/plugins/README.md) that use alternate methods. Protocols used by the SNMPDSC or DHCPLSS plugins have good support and usually can be used as a workaround.

> Alternatively, you can set up separate NetAlertX instances running on the subnets and synchronize the results into one instance with the [SYNC plugin](https://github.com/jokob-sk/NetAlertX/tree/main/front/plugins/sync).

You can also double check you are listing the correct subnets and interfaces. 

https://github.com/jokob-sk/NetAlertX/blob/main/docs/SUBNETS.md#network-interface-adapter

**Comment by Zerbaib** [comment]
i have created a new fresh vm et do a ping and a scan:
![image](https://github.com/user-attachments/assets/fac0d883-305a-4a2a-8bd6-5a820b5a33b9)
if this can help

**Comment by Zerbaib** [comment]
my vm as only a acces to 192.168.0.0/24 but the router 192.168.0.1/24 has acces to all network
si if i scan 192.168.1.0/24 it's normaly work no ?

**Comment by jokob-sk** [comment]
Hey, according to ChatGPT pinging a device doesn't mean taht arpscan has access to it as well:

> Yes, it can be expected in certain scenarios where ping works but arpscan fails. The difference in how they operate might explain the behavior:
> 
>  -   Ping: Uses the Internet Control Message Protocol (ICMP) to send Echo Request packets to a host and waits for an Echo Reply. This works if the target device is reachable and ICMP is not blocked by a firewall.
> 
>   -  ARPSCAN: Relies on the Address Resolution Protocol (ARP), which is used to map IP addresses to MAC addresses within the local network segment (Layer 2). arpscan sends ARP requests and listens for ARP replies to discover devices on the local subnet.
> 
> Possible reasons why arpscan is failing while ping works:
> 
>  - Network segmentation: If you're scanning a different subnet (e.g., through a router), ARP scanning might fail because ARP only works within a local subnet.
>  - VLANs: If the devices are on different VLANs, ARP discovery might not function unless you have special configurations like inter-VLAN routing.
>  - Firewall restrictions: Some devices might block or filter ARP requests or replies, especially if the network administrator has configured them for enhanced security.
>  - Promiscuous mode: arpscan requires network interfaces to be in promiscuous mode to capture all ARP responses. If your interface is not configured correctly, it may miss replies.
> 
> If you're working across different subnets or a routed network, ping would work because routers forward ICMP packets, while ARP doesn't traverse beyond the local network.

If you haven't please read this page (especially the VLANS section at the bottom): https://github.com/jokob-sk/NetAlertX/blob/main/docs/SUBNETS.md#vlans

I'm not an expert on VLANs (I don't use them) so according to chatGPT you can:

> 1. Enable Inter-VLAN Routing
> 
> For VLANs to communicate, you need to enable inter-VLAN routing on a router or Layer 3 switch. This allows traffic between VLANs, but ARP will still be constrained to local VLANs. The devices would need routing rules to communicate across subnets, but arp-scan wouldn't work natively across these VLANs.
> 
>     Steps:
>         Configure a Layer 3 switch or router to route traffic between VLANs.
>         Ensure that you have IP routing enabled between the subnets.
>         Use a tool like ping or other network discovery protocols like LLDP or Nmap to detect devices across VLANs.
> 
> 2. Scan Each VLAN Separately
> 
> You could connect to each VLAN individually from your machine and then run arpscan from within each VLAN. This approach involves switching your machine to the target VLAN for the scan, or using virtual interfaces on your machine.
> 
>     Steps:
>         Create virtual network interfaces on your system, each assigned to a different VLAN (if your network adapter and switch support it).
>         Alternatively, switch your physical network interface to each VLAN manually, then run arpscan within each VLAN.
>         To add a virtual interface in Linux with VLAN ID 10:
> 
>         bash
> 
> sudo ip link add link eth0 name eth0.10 type vlan id 10
> sudo ip addr add 192.168.10.100/24 dev eth0.10
> sudo ip link set up dev eth0.10


Lastly, NetAlertX supports complementing protocols that can be use to scan other subnets, e.g. via SNMP, dhcp.leases or having separate NetAlertX instances running on the subnets and using the Sync plugin. 

> If direct scans are not possible, you can use [supplementing plugins](https://github.com/jokob-sk/NetAlertX/blob/main/front/plugins/README.md) that use alternate methods. Protocols used by the SNMPDSC or DHCPLSS plugins have good support and usually can be used as a workaround.
> 
> Alternatively, you can set up separate NetAlertX instances running on the subnets and synchronize the results into one instance with the [SYNC plugin](https://github.com/jokob-sk/NetAlertX/tree/main/front/plugins/sync).

Please let me know if you get ARPSSCAN working with e.g.  Inter-VLAN Routing and how so I can include it in the docs.

**Comment by jokob-sk** [comment]
no reply for a week -> closing
